Objective-C/CBLQueryResultSet.mm

```diff
@@ -80,7 +81,8 @@ - (instancetype) initWithQuery: (CBLQuery*)query
             return nil;
         _query = query;
         _c4enum = e;
-        _context = (cbl::QueryResultContext*)(new cbl::QueryResultContext(query.database, e))->retain();
+        CBLCollection* c = [query.database defaultCollectionOrThrow];
+        _context = (cbl::QueryResultContext*)(new cbl::QueryResultContext(c, e))->retain();
```

Comment:
QueryResultContext still uses database not collection. Query is per database not collection.